### ISSUE TITLE
Lambda parameters are not checked for a number of error conditions

### ISSUE THREAD

**Issue by 333fred** [dotnet/roslyn#62960]
```cs
Delegate d = (ref TypedReference typedReference) => 1;
```

This should error with `error CS1601: Cannot make reference to variable of type 'TypedReference'`. It does not, and compiles successfully.

**Comment by Youssef1313** [comment]
Most symbols use `ParameterHelpers.MakeParameters` to create the parameter symbols, which is the centralized place for most parameter diagnostics.

Lambdas are not using this helper and are reporting diagnostics on their own.

I see this very error-prone. For example, with default parameters, we likely need to duplicate this check in the lambda symbol/binding:

https://github.com/dotnet/roslyn/blob/35fda5cf56445624cb0837b8b9ddbd9aa8f98b95/src/Compilers/CSharp/Portable/Symbols/Source/ParameterHelpers.cs#L638-L643

I think it is much better if the code can be refactored such that it uses `ParameterHelpers`.
